contracts/src/modules/DummyModule.sol

```diff
@@ -1,48 +1,100 @@
+
+  // address (Safe address) => DummyConfig
+  mapping(address => DummyConfig) public safeConfigs;
 
-  constructor(address _vault, uint256 _cadence) {
-    vault = _vault;
-    cadence = _cadence;
+  constructor(address _protocolManager) {
+    protocolManager = protocolManager;
   }
 
   ////////////////////////////////////////////////////////////////////////////
-  // PUBLIC: Owner - Config
+  // PUBLIC: VIEW
   ////////////////////////////////////////////////////////////////////////////
 
-  /// @notice Updates the vault erc-4626 address at which rewards will be process
-  /// @param _vaultAddr The new vault adress as target
-  function setVault(address _vaultAddr) external {
-    address oldVaultAddr = vault;
+  /**
+   * @notice A funtion that returns name of the plugin
+   * @return name string name of the plugin
+   */
+  function name() external view returns (string memory name) {
+    name = "dummy";
+  }
 
-    vault = _vaultAddr;
-    emit VaultAddressUpdated(_vaultAddr, oldVaultAddr);
+  /**
+   * @notice A function that returns version of the plugin
+   * @return version string version of the plugin
+   */
+  function version() external view returns (string memory version) {
+    version = "0.0.1";
   }
 
-  /// @notice Updates the cadenceat which rewards are processed by the keeper
-  /// @param _cadence The new cadence frequency in seconds.
-  function setCadence(uint256 _cadence) external {
-    uint256 oldCadence = cadence;
+  /**
+   * @notice A function that returns information about the type of metadata provider and its location.
+   *         For more information on metadata provider, refer to https://github.com/safe-global/safe-core-protocol-specs/.
+   * @return providerType uint256 Type of metadata provider
+   * @return location bytes
+   */
+  function metadataProvider()
+    external
+    view
+    returns (uint256 providerType, bytes memory location)
+  {
+    providerType = 0;
+    location = abi.encode(0);
+  }
+
+  /**
+   * @notice A function that indicates if the plugin requires root access to a Safe.
+   * @return requiresRootAccess True if root access is required, false otherwise.
+   */
+  function requiresRootAccess()
+    external
+    view
+    returns (bool requiresRootAccess)
+  {
+    requiresRootAccess = true;
+  }
+
+  ////////////////////////////////////////////////////////////////////////////
+  // PUBLIC: Manager - Config
+  ////////////////////////////////////////////////////////////////////////////
+
+  function setSafeConfig(address safe, DummyConfig calldata config) external {
+    safeConfigs[safe] = config;
+  }
 
-    cadence = _cadence;
-    emit CadenceUpdated(_cadence, oldCadence);
+  /**
+   * @notice Executes a Safe transaction
+   * @param manager Address of the Safe{Core} Protocol Manager.
+   * @param safe Safe account
+   * @param rootAccess Contains the set of actions to be done in the Safe transaction
+   */
+  function executeFromPlugin(
+    ISafeProtocolManager manager,
```

Comment:
should we not rewrite this to a `ISmartGardenManager`?

Reply:
the exec is happening from the plugin itself which moves into the manager and the manager calls internally the `execTransactionFromModuleReturnData` method